examples/chatgpt/src/chatgpt_bridge/actions/cutout.py

```diff
@@ -25,32 +25,47 @@ async def process(
     ctx: EditorAPIContext,
     stateid_input: StateID,
     background_color: str,
-    object_name: str,
+    prompt: str,
 ) -> tuple[Image.Image, StateID]:
-    # call infer-bbox
-    result_bbox = await ctx.call_async.infer_bbox(
+    # call detect
+    result_detect = await ctx.call_async.detect(
         state_id=stateid_input,
-        product_name=object_name,
+        prompt=prompt,
     )
-    if isinstance(result_bbox, ErrorResult):
-        raise ValueError(f"Cutout internal infer_bbox error: {result_bbox.error}")
-    bbox = result_bbox.bbox
-    app.logger.debug(f"{bbox=}")
-
-    # call segment
-    result_segment = await ctx.call_async.segment(
-        state_id=stateid_input,
-        bbox=bbox,
-    )
-    if isinstance(result_segment, ErrorResult):
-        raise ValueError(f"Cutout internal segment error: {result_segment.error}")
-    stateid_segment = result_segment.state_id
-    app.logger.debug(f"{stateid_segment=}")
+    if isinstance(result_detect, ErrorResult):
+        raise ValueError(f"Cutout internal detect error: {result_detect.error}")
+    detections = result_detect.results
+    app.logger.debug(f"{detections=}")
+
+    # call segment on each detection
```

Comment:
nit: perhaps you want to mimic:

https://github.com/finegrain-ai/finegrain-python/blob/df07310daa078ec54b7ed31988b096142d14c3ea/examples/discord/src/finegrain_discord/bot.py#L163-L170

Also, do you want to factorize the logic in a multi-segment or something like in the Discord bot? (make it easier to re-use across skills) - didn't think too much about it